src/main/java/cn/nukkit/block/BlockRespawnAnchor.java

```diff
@@ -0,0 +1,214 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package cn.nukkit.block;
+
+import cn.nukkit.Player;
+import cn.nukkit.api.PowerNukkitOnly;
+import cn.nukkit.api.Since;
+import cn.nukkit.blockproperty.BlockProperties;
+import cn.nukkit.blockproperty.IntBlockProperty;
+import cn.nukkit.event.block.BlockExplosionPrimeEvent;
+import cn.nukkit.item.Item;
+import cn.nukkit.item.ItemTool;
+import cn.nukkit.lang.TranslationContainer;
+import cn.nukkit.level.Explosion;
+import cn.nukkit.level.GameRule;
+import cn.nukkit.level.Level;
+import cn.nukkit.level.Sound;
+import cn.nukkit.utils.BlockColor;
+import cn.nukkit.utils.TextFormat;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+/**
+ * @author joserobjr
+ * @since 2020-10-06
+ */
+@PowerNukkitOnly
+@Since("1.4.0.0-PN")
+public class BlockRespawnAnchor extends BlockMeta {
+    @PowerNukkitOnly
+    @Since("1.4.0.0-PN")
+    public static final IntBlockProperty RESPAWN_ANCHOR_CHARGE = new IntBlockProperty("respawn_anchor_charge", true, 4);
+    
+    @PowerNukkitOnly
+    @Since("1.4.0.0-PN")
+    public static final BlockProperties PROPERTIES = new BlockProperties(RESPAWN_ANCHOR_CHARGE);
+
+    @Override
+    public int getId() {
+        return RESPAWN_ANCHOR;
+    }
+    
+    @Since("1.4.0.0-PN")
+    @PowerNukkitOnly
+    @Nonnull
+    @Override
+    public BlockProperties getProperties() {
+        return PROPERTIES;
+    }
+
+    @Override
+    public String getName() {
+        return "Respawn Anchor";
+    }
+
+    @Override
+    public boolean onActivate(@Nonnull Item item, @Nullable Player player) {
+        int charge = getCharge();
+        if (item.getBlockId() == BlockID.GLOWSTONE_BLOCK && charge < RESPAWN_ANCHOR_CHARGE.getMaxValue()) {
+            if (player == null || !player.isCreative()) {
+                item.count--;
+            }
+
+            setCharge(charge + 1);
+            getLevel().addSound(this, Sound.RESPAWN_ANCHOR_CHARGE);
+            return true;
+        }
+        
+        if (player == null) {
+            return false;
+        }
+        
+        return attemptToSetSpawn(player);
+    }
+
+    @Since("1.4.0.0-PN")
+    @PowerNukkitOnly
+    protected boolean attemptToSetSpawn(@Nonnull Player player) {
+        if (this.level.getDimension() != Level.DIMENSION_OVERWORLD) {
```

Comment:
You accidentally mistook the Respawn Anchor for the bed. The bed is to explode in other dimensions and the respawn anchor in the overworld.